### PR TITLE
Create organization

### PR DESCRIPTION
**Pull request by alonoslav**

### CODE REVIEW

File: both/lib/router-config.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  } else if (userId) {
+    var isOwner = Roles.userIsInRole(userId, ['owner']);
+    organization = AntiToggl.organizations.isOrgOwner(userId);
+    if (isOwner && organization) {
+      return this.next();
+    }else if (isOwner && !organization){
+      Router.go('createOrg');
+      return this.next();
+    }else{
+      Router.go('home');
```
Comment (alonoslav) [inline both/lib/router-config.js]: @hhalka go('signIn') here

File: both/lib/router-config.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+      return this.next();
+    }
+    return this.next();
```
Comment (alonoslav) [inline both/lib/router-config.js]: @hhalka we don't need this

File: both/lib/router.js [hunk 3]
```diff
@@ -23,3 +23,22 @@ Router.route('/', {
   name: 'home',
   template: 'home',
 });
+
+Router.route('organizations/:_id', {
```
Comment (alonoslav) [inline both/lib/router.js]: @hhalka can you move routers that belong to organizations to a separate file and rename this file to `home.js`.
also, you should to create a directory where we will place our routes

File: both/lib/router.js [hunk 4]
```diff
@@ -23,3 +23,22 @@ Router.route('/', {
...
+  name: 'organization',
+  waitOn: function() {
+    return Meteor.subscribe('organizations');
```
Comment (alonoslav) [inline both/lib/router.js]: @hhalka you were subscribed on organizations in startup. You should to remove subscriptions in waitOn

File: both/methods/create-org-method.js [hunk 5]
```diff
@@ -0,0 +1,10 @@
+Meteor.methods({
+  createOrganization: function(organization) {
+    if(Organizations.findOne(organization)){
+      throw new Meteor.Error('This organization already exists');
+    }else{
+      id = Organizations.insert(organization);
```
Comment (alonoslav) [inline both/methods/create-org-method.js]: @hhalka not defined variable

File: client/view/app/create-org.html [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+      <div class='at-form'>
+        <div class='at-title'>
+          <h3>Create your organization</h3>
+        </div>
+        <div class='at-pwd-form'>
+          <form role='form' id='register-org-form'>
+            <fieldset>
+              <div class='at-input form-group'>
+                <label class='control-label' for='at-field-name'>Your organization title</label>
+                <input type='text' class='form-control' id='organizationName' placeholder='Title'/>
```
Comment (alonoslav) [inline client/view/app/create-org.html]: @hhalka add require attribute here

File: client/view/app/create-org.js [hunk 7]
```diff
@@ -0,0 +1,21 @@
+Template.createOrg.events({
+  'submit form' : function(event, template) {
+    event.preventDefault();
+
+    var orgName = event.target.organizationName.value;
```
Comment (alonoslav) [inline client/view/app/create-org.js]: @hhalka you should to trim text from inputs

File: client/view/app/organization.html [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+        <div class='orgEmail'>
+          <h4>Main e-mail</h4>
+
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<template name='logoTemplate'>
```
Comment (alonoslav) [inline client/view/app/organization.html]: @hhalka create one file for one template. If templates are nested, create nested folders

File: server/publications/organizations.js [hunk 9]
```diff
@@ -1,3 +1,3 @@
-Meteor.publish('organizations', function(){
-	return Organizations.find();
-})
\ No newline at end of file
+Meteor.publish('organizations', function() {
+	return Organizations.find({owners: this.userId});
```
Comment (alonoslav) [inline server/publications/organizations.js]: @hhalka here you should to check all organizations where current user is, not only owner. In user document there is a field organizations with array of organization IDs. you should publish them too

File: both/methods/create-org-method.js [hunk 10]
```diff
@@ -0,0 +1,11 @@
+Meteor.methods({
+  createOrganization: function(organization) {
+    if(Organizations.findOne(organization)){
+      throw new Meteor.Error('This organization already exists');
+    }else{
+      var organizationId = Organizations.insert(organization);
+      Meteor.users.update(this.userId, {$push: {organizations: organizationId}});
+      return Organizations.findOne({_id: organizationId});
```
Comment (alonoslav) [inline both/methods/create-org-method.js]: @hhalka you should to return only `organizationId`. In router you use this parameter for organization subscription

File: client/view/app/organization.js [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+
+Template.logoTemplate.events({
+  'click .deleteFileButton': function(event) {
+    Logos.remove({_id: this._id});
+  },
+  'change .uploadLogo': function(event, template) {
+    var files = event.target.files;
+    for (var i = 0; i < files.length; i++) {
+      Logos.insert(files[i], function (err, fileObj) {
+        if(fileObj) {
```
Comment (alonoslav) [inline client/view/app/organization.js]: @hhalka the same if else we can replace by `console.log(!!fileObj)`. this operator converts any type to boolean

File: both/methods/create-org-method.js (hunk 10 above)
Comment (hhalka) [inline both/methods/create-org-method.js]: I don't know exactly, but: 
Router.route('organizations/:_id', {
  name: 'organization',
  data: function() {
    return Organizations.findOne(**this.params._id**);
  },

so it looks like the whole object needed

Comment (alonoslav) [inline both/methods/create-org-method.js]: @hhalka if you pass all organization object into router, this.params._id will be equal to this object, not to _id only

**Pull request merged**